src/System.Private.CoreLib/shared/System/Decimal.DecCalc.cs

```diff
@@ -2013,7 +2015,7 @@ internal static unsafe void VarDecDiv(ref DecCalc d1, ref DecCalc d2)
                 bool fUnscale = false;
                 uint ulTmp;
 
-                if (d2.High == 0 && d2.Mid == 0)
+                if ((d2.High | d2.Mid) == 0)
```

Comment:
This would be an interesting JIT optimization if it's actually faster. cc @AndyAyersMS

Reply:
The jit doesn't do anything like this today.

Combining tests like this -- where the expressions are side effect free and low cost -- can often pay off. But it is not always a win. It depends on the predictability of the branches and what else is going on in the neighborhood.

For instance if `d2.High` is usually nonzero, perf might get a bit worse by combining the checks this way. 

But I'm guessing that for decimals the high part is likely to be zero and this is a pretty safe win.